panther_lights/src/lights_driver_node.cpp

```diff
@@ -109,6 +107,12 @@ void DriverNode::ClearLEDs()
 
 void DriverNode::ToggleLEDControl(const bool enable)
 {
+  RCLCPP_DEBUG(
+    this->get_logger(), "Calling service to toggle LED control to '%s'.",
+    enable ? "true" : "false");
+
+  led_control_status_ = LEDControlStatus::PENDING;
```

Comment:
If there is no response from the server, the client node is locked in a `PENDING` state.

Reply:
You suggest adding a timeout for server response?

Reply:
We would need to choose either checking timeout (most likely on a timer?) or sending the request until the control is granted without checking the PENDING state. The first option is way more elegant but demands more changes.